jetbrains-core/src/software/aws/toolkits/jetbrains/core/gettingstarted/OuterPanel.kt

```diff
@@ -0,0 +1,423 @@
+                            applyToComponent {
+                                putClientProperty(DarculaButtonUI.DEFAULT_STYLE_KEY, true)
+                            }
+                        }
+                        topGap(TopGap.SMALL)
+                    }
+
+                    row {
+                        label(message("codewhisperer.gettingstarted.panel.licence_comment")).applyToComponent { foreground = Paneltext.TEXT_FONTCOLOR }
+                    }
+                    row {
+                        browserLink(message("aws.onboarding.getstarted.panel.login_with_iam"), url = CODEWHISPERER_LEARN_MORE_URI)
+                    }
+                }
+            }
+        )
+
+        border = IdeBorderFactory.createRoundedBorder().apply {
+            setColor(CodeWhispererColorUtil.POPUP_BUTTON_BORDER)
+
+            roundedCorners
+            preferredSize = Dimension(Paneltext.PANEL_WIDTH, Paneltext.PANEL_HEIGHT)
+        }
+    }
+}
+
+class ShareFeedbackInGetStarted : DumbAwareAction() {
+
+    override fun actionPerformed(e: AnActionEvent) {
+        runInEdt {
+            FeedbackDialog(DefaultProjectFactory.getInstance().defaultProject).show()
+        }
+    }
+}
+
+class CodeWhispererAuthBullets(private var panelTitle: String) : BorderLayoutPanel() {
+
+    init {
+        addToCenter(
+            panel {
+                indent {
+
+                    row {
+                        label(panelTitle).applyToComponent { font = JBFont.h3() }
+                    }
+                    row {
+                        text(Paneltext.COMMIT_ICON)
+                        panel {
+                            row(message("iam_identity_center.name")) {
+                            }
+                                .rowComment(message("aws.onboarding.getstarted.panel.idc_row_comment_text"))
+                        }
+                    }
+
+                    row {
+                        text(Paneltext.COMMIT_ICON)
+                        panel {
+                            row(message("aws_builder_id.service_name")) {
+                            }.rowComment("Need to insert tagline")
+                        }
+                    }
+                    row {
+                        text(Paneltext.CANCEL_ICON)
+                        panel {
+                            row(message("settings.credentials.iam")) {
+                            }
+                                .rowComment(message("aws.getstarted.auth.panel.notSupport_text"))
+                        }
+                    }
+                }
+            }
+        )
+
+        border = IdeBorderFactory.createRoundedBorder().apply {
+            setColor(CodeWhispererColorUtil.POPUP_BUTTON_BORDER)
+            preferredSize = Dimension(Paneltext.PANEL_WIDTH, Paneltext.BULLET_PANEL_HEIGHT)
+        }
+    }
+}
+
+class ResourceExplorerAuthBullets(private var panelTitle: String) : BorderLayoutPanel() {
```

Comment:
these bullet panels are basically all the same
can we simplify the implementation?

Reply:
Done